src/polymer/expression-scanner.ts

```diff
@@ -493,7 +493,7 @@ export function parseExpressionInJsStringLiteral(
   };
   const sourceRangeForLiteral = document.sourceRangeForNode(stringLiteral)!;
 
-  if (!babel.isLiteral(stringLiteral)) {
+  if (!astValue.isLiteralExpression(stringLiteral)) {
```

Comment:
What if, instead of this check, we wrote it more like:

```typescript
const expressionText = astValue.expressionToValue(stringLiteral);
if (expressionText === undefined) {
  // emit the unanalyzable expression warning
  return result;
}
if (typeof expressionText !== 'string') {	   
  // emit invalid expression warning
  return result;
}
```

Because `astValue.expressionToValue` already has a good idea of what it can and can't handle.

Reply:
yup i was thinking the same. i dont think i need the new function i added. because if we pass a binary expression we don't understand, we already get `undefined` back